packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/SplitButtonWidget.tsx

```diff
@@ -0,0 +1,352 @@
+import HelpOutlineOutlined from '@material-ui/icons/HelpOutlineOutlined';
+import React, { useContext, useEffect, useRef, useState } from 'react';
+import { addWidgetMutation } from './FormDescriptionEditorEventFragment';
+import {
+  GQLAddWidgetInput,
+  GQLAddWidgetMutationData,
+  GQLAddWidgetMutationVariables,
+  GQLAddWidgetPayload,
+  GQLErrorPayload,
+} from './FormDescriptionEditorEventFragment.types';
+import { SplitButtonWidgetState } from './SplitButton.type';
+import { SplitButtonWidgetProps } from './WidgetEntry.types';
+
+const isErrorPayload = (payload: GQLAddWidgetPayload): payload is GQLErrorPayload =>
+  payload.__typename === 'ErrorPayload';
+
+const actionStyle = (theme: Theme, style: GQLButtonStyle): React.CSSProperties => {
+  const actionStyle: React.CSSProperties = {
+    backgroundColor: style.backgroundColor ? getCSSColor(style.backgroundColor, theme) : theme.palette.primary.light,
+    color: style.foregroundColor ? getCSSColor(style.foregroundColor, theme) : 'white',
+    fontSize: style.fontSize ? style.fontSize : undefined,
+    fontStyle: style.italic ? 'italic' : undefined,
+    fontWeight: style.bold ? 'bold' : undefined,
+    textDecorationLine:
+      style.underline || style.strikeThrough
+        ? getTextDecorationLineValue(style.underline, style.strikeThrough)
+        : undefined,
+  };
+
+  return actionStyle;
+};
+
+const imageStyle = (theme: Theme, iconOnly: boolean): React.CSSProperties => {
+  return {
+    marginRight: iconOnly ? theme.spacing(2) : theme.spacing(0),
+    width: theme.spacing(2),
+    height: theme.spacing(2),
+  };
+};
+
+const useStyles = makeStyles<Theme>((theme) => ({
+  bottomDropArea: {
+    display: 'flex',
+    flexDirection: 'column',
+    alignItems: 'center',
+    justifyContent: 'center',
+    backgroundColor: 'whitesmoke',
+    borderRadius: '10px',
+    color: 'gray',
+    height: '30px',
+    width: '100%',
+  },
+  helpOutlineOutlined: {
+    marginLeft: theme.spacing(1),
+    fontSize: theme.spacing(2),
+  },
+  dragOver: {
+    borderWidth: '1px',
+    borderStyle: 'dashed',
+    borderColor: theme.palette.primary.main,
+  },
+  selected: {
+    color: theme.palette.primary.main,
+  },
+  containerAndLabel: {
+    margin: 0,
+    padding: 0,
+    borderWidth: 1,
+    borderColor: 'gray',
+    borderStyle: 'solid',
+    borderRadius: 0,
+  },
+  propertySectionLabel: {
+    display: 'flex',
+    flexDirection: 'row',
+    alignItems: 'center',
+  },
+}));
+
+export const SplitButtonWidget = ({ widget, editingContextId, representationId }: SplitButtonWidgetProps) => {
+  const initialState: SplitButtonWidgetState = {
+    buttonLabel: widget.buttonLabel,
+    selected: false,
+    open: false,
+    selectedIndex: 0,
+    message: '',
+  };
+
+  const [state, setState] = useState<SplitButtonWidgetState>(initialState);
+  const { httpOrigin } = useContext<ServerContextValue>(ServerContext);
+  const anchorRef = React.useRef<any>(null);
```

Comment:
Could you set the real possible types of the `useRef` here please and remove the `React.`? Should be something like `<HTMLDivElement | null>` instead of `any`.  Could you also rename it into something more explicit? Here it is difficult to make the difference between `ref` and `anchorRef`.